# HeNe Laser Lab Notes

## 2019-02-06

SPECTROSCOPY
- Set up lock-in amplifier at phase 0 for 0-90 and 0 for 0-180, Sensitivity 30 mV
- Frequency of chopper is 107 Hz +/- 1
- Turned dial on michromotor to see at which wavelengths the signal on oscilloscope drops and when the lock-in amplifier voltage gauge changes.
    
From turning the chopper, we  observed ripples in the voltage reading of the photomultiplier at 599 nm and 615 nm.


| Wavelength | voltage | modulated | measurement on lock-in |
|-|-|-|-|
|505 nm |  -633 mV | - | - |
| 538 nm | -130 mV | - | - |
| 581 nm | -250 mV | - | - |
| 587 nm | -146 mV | - | - |
| 590 nm | -660 mV | - | - |
| 593 nm | - 1.05 V | - | - |
| 599 nm | -400 mV | modulated | -5dB |
| 615 nm | -380 mV | modulated | -6 dB |
| 619 nm | -386 mV | - | - |
| 632 nm | -150 mV | - | - |
| 645 nm | -520 mV | modulated | +0.8 dB |
| 785 nm | -300 mV | - | - |


### Repeated measurements at Sensitivity at 10dB

__bold__ denotes that this wavelength did not show up at previous sensitivity 

| Wavelength | voltage | measurement on lock-in |
|-|-|-|
| 505 nm |  -633 mV | - |
| 538 nm | -130 mV | - |
| 581 nm | -250 mV | +1 dB  |
| 587 nm | -146 mV | +0.8 dB |
| 590 nm | -660 mV | +3 dB |
| 593 nm | - 1.05 V | +.5 dB |
| 599 nm | -400 mV | ~~pos (gauge overloaded)~~ probably recorded wrong. It should be negative. |
| __608 nm__ | -83 mV | +0.4 dB |

### Sensitivity was changed to 3 dB

|wavelength|voltage|measurement on lock-in|
|-|-|-|
|__613 nm__ | -190 mV | neg |
|615 nm | -380 mV | neg|
|619 nm | -386 mV | pos|
|__622 nm__| -120 mV | pos|
|632 nm | -150 mV | -.3 dB * Needle moving back and forth . Probably just noise. |
|__639 nm__ | -105 mV | pos|
|645 nm | -520 mV | pos|
|__657 nm__ |  -130 mV | pos|
|__666 nm__ | -50 mV | pos|
|__674 nm__ | -130 mV | neg|

Higher sensitivity of the lock-in amplifier enables more wavelengths to be detected. Sensitivity of 3 mV worked best, as sensitivity of 1 mV picked up too much noise.
    
We would have expected the 632 nm line to have the largest absolute voltage reading since we know that the primary emission wavelength of the HeNe laser is 632.8 nm.
But it seems that the largest absolute reading is 645 nm.| 

## 2019-02-13

Discussed with TAs / prof today about the wavelengths.
We have been told that the readings on our monochromator are likely not correct and have some offset in the readings.
This is consistent with our observations last day that the primary emission wavelength does not seem to be at 632.8 nm.

Today the laser was also unaligned. Following the instructions in the video, we used the following alignment steps:

1. loosen the base mount
2. raise the vertical alignment to be higher than where it should be aligned
3. wiggle the base left and right while slowly lowering the mirror alignment until flashes are seen

This process took about 7 minutes.

We experimented with adjusting the position of the mirror.
At some positions the laser will stop emitting. One such distance was approximately 36.5 cm
for mirror-mirror separation. Another is at about 60.5cm for mirror-mirror separation.

These measurements are not very accurate since the mirror is positioned quite far away from
the point of measurement

| positions of minimal intensity (cm) |
|-|
| ~~36.5~~ |
| ~~39.0~~ |
| ~~40.5~~ |
| ~~41.0~~ |
| ~~42.0~~ |
| ~~44.0~~ | 
| ~~60.5~~ | 

The readings above went to minimal intensity simply because we screwed up the horizontal alignment too much
when moving the mirror. We can't actually find any points where the laser should not work anymore.

This makes sense since manual adjustments are unlikely to be precise enough to find a minimum.
This is supported by the calculation below which computes the mode number for the laser, with a slight perturbation of 10 $\mu m$.
This perturbation causes a mode number difference of 32, which is a lot.


In [1]:
lambda_c = 632.8 * (10**-9)
L1 = 0.6
L2 = 0.60001
n1 = 2 * L1 / lambda_c
n2 = 2 * L2 / lambda_c
print(n1)
print(n2)

1896333.7547408342
1896365.3603034136


# 2019-02-27

## Repeated measurements at Sensitivity 3 dB after calibration

Time constant: TC x 0.1 BW x 10

|wavelength|voltage|measurement on lock-in| comments | 
|-|-|-|-| 
| 393 nm | -430 mV | No change | - | 
| 407 nm | -70 mV | No change | - | 
| 451 nm | -384 mV | No change | - | 
| 475 nm | -175 mV | No change | - | 
| 496 nm | -150 mV | No change | - | 
| 506 nm | -520 mV | No change | - | 
| 508 nm | -80 mV | No change | - |  
| 509 nm | -40 mV | No change | - | 
| 512 nm | -40 mV | No change | - | 
| 538 nm | -86 mV | No change | - | 
| 581 nm | -186 mV | +2 dB | 3 dB sensitivity | 
| 587 nm | -110 mv | +4 dB | 1 dB sensitivity | 
| 590 nm | -510 mv | +5 dB | 3 dB sensitivity | 
| 593 nm | -800 mV | - | Oscillating a lot. Probably just noise. | 
| 595 nm | -60 mV | +5 dB | 1 dB sensitivity | 
| 599 nm | -280 mV | -9 dB | 10 dB sensitivity | 
| 608 nm | -66 mV | +3 dB | 1 dB sensitivity | 
| 613 nm |  -150 mV | -6 dB |  3 dB sensitivity | 
| 615 nm | -260 mV | -3 dB | 30 dB sensitivity | 
| 620 nm | -285 mV | +3 dB | 3 dB sensitivity | 
| 622 nm | -100 mV | +5 dB | 1 dB sensitivty | 
| 627 nm | -40 mV | No change | - | 
| 632 nm | -100 mV | - | Small oscillations. Probably just noise. | 
| 639 nm | -80 mV | +2 dB | 1 dB sensitivity |  
| 644 nm | -100 mV | +4-8 dB | 0.3 dB sensitivity; possibly noise | 
| 646 nm | -360 mV | +4-+5 dB |  3 dB sensitivity | 
| 656 nm | -100 mV | +2-+4 dB |  1 dB sensitivity | 
| 665 nm | -40 mV | +1-+3 dB |  1 dB sensitivity | 
| 674 nm | -100 mV | -6 dB | 3 dB sensitivity |
| 678 nm | -30 mV | No change | - | 
| 699 nm | -20 mV | 1.5 dB | 0.3 dB sensitivity; increased time constant to 400| 
| 710 nm | -25 mV | +2 dB | 0.3 dB sensitivity | 
| 785 nm | -222 mV | - | Small oscillations. Probably just noise. | 
| 814 nm | -30 mV | No change |  - | 


## Monochromator calibration

| Monochromator wavelength (nm)| Expected wavelength nearby (nm)|Photomultiplier voltage (mv)|
|-|-|-|
|409|404.656|-600|
|440|435.833|-5000|
|550|546.074|-3500|
|582|576.960|-80|
|584|579.066|-80|

__note__ somehow we did not detect any wavelength that we expected to correspond with the 407.783 nm line.

Based on the link: https://www.newport.com/t/calibrating-cornerstone-monochromators-and-ms260i-spectrographs, a monochromator should have a calibration of offset as well a slope. 

### Code for linear fit

In [14]:
import numpy as np

rows = []
with open('mercury-argon.txt', 'r') as f:
    for line in f:
        row = []
        for column in line.split('|'):
            try:
                row.append(float(column))
            except:
                continue
        rows.append(row)
        
x = []
y = []
for row in rows:
    x.append(row[0])
    y.append(row[1]-row[0])
calib = np.poly1d(np.polyfit(x, y, 1))
print(calib)

 
-0.003198 x - 2.841


### Linear fit:

__slope__: 0.003198

__offset__: 2.841


### Code to create table using monochromator calibration

In [19]:
with open('hene.txt', 'r') as f_in:
    with open('hene-calib.txt', 'w') as f_out:
        for line in f_in:
            columns = line.split('|')
            wavelength = float(columns[1].split()[0])
            columns[1] =  '{0:g}'.format(round(wavelength + calib(wavelength)))
            f_out.write('|'.join(columns))

### Modified HeNe data using calibrated wavelengths

|wavelength (nm)|voltage|measurement on lock-in| comments | 
|-|-|-|-| 
|389| -430 mV | No change | - |
|403| -70 mV | No change | - |
|447| -384 mV | No change | - |
|471| -175 mV | No change | - |
|492| -150 mV | No change | - |
|502| -520 mV | No change | - |
|504| -80 mV | No change | - |
|505| -40 mV | No change | - |
|508| -40 mV | No change | - |
|533| -86 mV | No change | - |
|576| -186 mV | +2 dB | 3 dB sensitivity |
|582| -110 mv | +4 dB | 1 dB sensitivity |
|585| -510 mv | +5 dB | 3 dB sensitivity |
|588| -800 mV | - | Oscillating a lot. Probably just noise. |
|590| -60 mV | +5 dB | 1 dB sensitivity |
|594| -280 mV | -9 dB | 10 dB sensitivity |
|603| -66 mV | +3 dB | 1 dB sensitivity |
|608|  -150 mV | -6 dB |  3 dB sensitivity |
|610| -260 mV | -3 dB | 30 dB sensitivity |
|615| -285 mV | +3 dB | 3 dB sensitivity |
|617| -100 mV | +5 dB | 1 dB sensitivty |
|622| -40 mV | No change | - |
|627| -100 mV | - | Small oscillations. Probably just noise. |
|634| -80 mV | +2 dB | 1 dB sensitivity |
|639| -100 mV | +4-8 dB | 0.3 dB sensitivity; possibly noise |
|641| -360 mV | +4-+5 dB |  3 dB sensitivity |
|651| -100 mV | +2-+4 dB |  1 dB sensitivity |
|660| -40 mV | +1-+3 dB |  1 dB sensitivity |
|669| -100 mV | -6 dB | 3 dB sensitivity |
|673| -30 mV | No change | - |
|694| -20 mV | 1.5 dB | 0.3 dB sensitivity; increased time constant to 400|
|705| -25 mV | +2 dB | 0.3 dB sensitivity |
|780| -222 mV | - | Small oscillations. Probably just noise. |
|809| -30 mV | No change |  - |
